src/Microsoft.AspNet.Mvc.TagHelpers/JavaScriptUtility.cs

```diff
@@ -68,6 +68,32 @@ internal static string PrepareFormatString(string input)
         }
 
         /// <summary>
+        /// Encodes a .NET string array for safe use as a JavaScript array literal, including inline in an HTML file.
+        /// </summary>
+        internal static string JavaScriptArrayEncode(IEnumerable<string> values)
+        {
+            var result = new StringBuilder();
+            result.Append("[");
+
+            var firstAdded = false;
+
+            foreach (var value in values)
+            {
+                if (firstAdded)
+                {
+                    result.Append(",");
```

Comment:
To avoid additional appends you can append the first one prior to this loop and then skip the first item in this loop and append every value as `result.AppendFormat(",\"{0}\"", JavaScriptStringEncode(value));`